### DISCUSSION TITLE
How to structure large scale deployments of thousands of services

### DISCUSSION THREAD

**Discussion by fdervisi** [aws/aws-cdk#18997] [General]
Dear Experts,

I am looking for guidance or just brainstorming on how an AWS CDK project should be structured to support 1000 or even 10,000 services, across multiple accounts and regions?

What is the strategy here?

- Aggregate individual services like VPC, subnets EC2 into stacks and use them in loops?
- L4 constructs with loops?
- NestedStacks?

How are the variable for the services managed? What hierarchy?

Does anyone have experience with deployments like this and can share some experience?

**Comment by indrora** [comment]
At this sort of scale, the best approach depends on the particular situation and requirements. There is no one single prescriptive solve-all solution.
